Formula/t/three-body.rb

```diff
@@ -0,0 +1,18 @@
+class ThreeBody < Formula
+  desc "三体编程语言 Three Body Language written in Rust"
+  homepage "https://github.com/rustq/3body-lang"
+  url "https://github.com/rustq/3body-lang/releases/download/0.3.0-beta/3body-lang.tar.gz"
+  sha256 "d381e6e21b26f461b767b52d189545803a20e34618da1cc235be12bd6790eee1"
+  license "MIT"
+  head "https://github.com/rustq/3body-lang.git", branch: "main"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_equal "0.3.0", shell_output("#{bin}/3body -v").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.